File: sdk/android/story-sdk/src/main/kotlin/com/openstory/sdk/internal/ui/StoryViewerTapRegion.kt
package com.openstory.sdk.internal.ui

internal enum class StoryViewerTapDirection {
    BACKWARD,
    FORWARD,
}

internal fun storyViewerTapDirection(
    tapX: Float,
    width: Int,
): StoryViewerTapDirection {
    return if (tapX < width * STORY_VIEWER_BACKWARD_TAP_WIDTH_RATIO) {
        StoryViewerTapDirection.BACKWARD
    } else {
        StoryViewerTapDirection.FORWARD
    }
}

private const val STORY_VIEWER_BACKWARD_TAP_WIDTH_RATIO = 0.25f

File: sdk/flutter/ios/open_story_flutter/Sources/open_story_flutter/OpenStorySDK/Internal/UI/StoryViewerTapRegion.swift
import Foundation

internal enum StoryViewerTapDirection {
    case backward
    case forward
}

internal func storyViewerTapDirection(
    tapX: Double,
    width: Double
) -> StoryViewerTapDirection {
    if tapX < width * storyViewerBackwardTapWidthRatio {
        return .backward
    }

    return .forward
}

private let storyViewerBackwardTapWidthRatio = 0.25
